apps/ejabberd/src/mongoose_api_backend.erl

```diff
@@ -78,64 +78,74 @@ rest_terminate(_Req, _State) ->
 allow_missing_post(Req, State)->
     {true, Req, State}.
 
+delete_resource(Req, State) ->
+    handle_delete(Req, State).
+
 %%--------------------------------------------------------------------
 %% internal callbacks
 %%--------------------------------------------------------------------
 
 to_json(Req, State) ->
-    handle_request(mongoose_api_json, Req, State).
+    handle_get(mongoose_api_json, Req, State).
 
 %%--------------------------------------------------------------------
 from_json(Req, State) ->
-    handle_body(mongoose_api_json, Req, State).
+    handle_post(mongoose_api_json, Req, State).
 
 %%--------------------------------------------------------------------
 %% internal funs
 %%--------------------------------------------------------------------
 
-handle_request(Serializer, Req, #backend_state{bindings=Bindings, command_category = Name}=State) ->
+handle_get(Serializer, Req, #backend_state{bindings=Bindings, command_category = Name}=State) ->
     %% should we introduce filtering not only by category?
     Commands = ?COMMANDS_ENGINE:list(admin, Name),
-    [Command] = [C || C <- Commands, ?COMMANDS_ENGINE:action(C) =:= read],
+    [Command] = [C || C <- Commands, (?COMMANDS_ENGINE:action(C) =:= read) ],
```

Comment:
There is a plan to make `mongoose_commands` module implement filtering by `category` and `action` and then those repeated lines will be compressed to something one line